ballerina/tests/timestamp-test.bal

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/sql;
+import ballerina/time;
+
+@test:Config {
+    groups: ["timestamp"]
+}
+public function testStringTimestamp() returns error? {
+    Client dbClient = check new (host, user, password, database = proceduresDatabase);
+
+    string datetimetz = "2021-07-21T12:00:00+03:30";
+    sql:ExecutionResult ret = check dbClient->execute("INSERT INTO DateTimeTypes (ROW_ID, TIMESTAMPTZ_TYPE) " +
+                                                      "VALUES(11, '"+datetimetz+"')");
+
+    InOutParameter rowIdInoutValue = new (11);
+    sql:DateTimeValue datetimetzValue = new();
+    InOutParameter datetimetzInoutValue = new (datetimetzValue);
+
+    sql:ParameterizedCallQuery sqlQuery =
+        `
+        CALL TimestampProcedure (${rowIdInoutValue}, ${datetimetzInoutValue});
+	    `;
+    sql:ProcedureCallResult result = check dbClient->call(sqlQuery, []);
+
+    test:assertEquals(check datetimetzInoutValue.get(time:Utc), check time:utcFromString(datetimetz),
+                      "Retrieved date time with timestamp does not match.");
+}
+
+@test:Config {
+    groups: ["timestamp"]
+}
+public function testStringParamTimestamp() returns error? {
+    Client dbClient = check new (host, user, password, database = proceduresDatabase);
```

Comment:
Write a util class which will take an initialised client and execute the query and returns the procedurecallresult

Reply:
Fixed in 0479e8d